File: clients/rust/model/src/weapon_params.rs
use crate::*;
#[derive(Clone, Debug, trans::Trans)]
pub struct WeaponParams {
    pub magazine_size: i32,
    pub fire_rate: f64,
    pub reload_time: f64,
    pub min_spread: f64,
    pub max_spread: f64,
    pub recoil: f64,
    pub aim_speed: f64,
    pub bullet: BulletParams,
    pub explosion: Option<ExplosionParams>,
}

File: clients/haskell/src/RAIC/Model/WeaponParams.hs
module RAIC.Model.WeaponParams where

import           GHC.Generics               (Generic)
import           RAIC.Model.BulletParams    (BulletParams)
import           RAIC.Model.ExplosionParams (ExplosionParams)
import           RAIC.StreamWrapper         (Trans)

data WeaponParams = WeaponParams {
  magazine_size :: Int,
  fire_rate     :: Double,
  reload_time   :: Double,
  min_spread    :: Double,
  max_spread    :: Double,
  recoil        :: Double,
  aim_speed     :: Double,
  bullet        :: BulletParams,
  explosion     :: Maybe ExplosionParams
} deriving (Generic, Show)

instance Trans WeaponParams
